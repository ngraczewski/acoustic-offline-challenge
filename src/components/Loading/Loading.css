@import '../../css/variables.css';

:root {
  --loading-indicator-size: 50px;
}

@keyframes pulse {
  50% {
    width: var(--loading-indicator-size);
    height: var(--loading-indicator-size);
    opacity: 100%;
    margin: 0;
    margin-right: var(--spacing);
  }

  0%,
  100% {
    --size-ratio: 0.2;
    --margin: calc(
      var(--loading-indicator-size) * 0.5 * (1 - var(--size-ratio))
    );
    --size: calc(var(--loading-indicator-size) * var(--size-ratio));
    width: var(--size);
    height: var(--size);
    opacity: 20%;
    margin: var(--margin);
    margin-right: calc(var(--margin) + var(--spacing));
  }
}

.loading {
  display: flex;
  justify-content: center;
}

.loading__message {
  font-weight: 500;
  display: flex;
  align-items: center;
}

.loading__message::before {
  content: '';
  background-color: var(--primary-color);
  width: var(--loading-indicator-size);
  height: var(--loading-indicator-size);
  border-radius: 50%;
  display: block;
  animation: pulse 1s infinite;
  box-shadow: var(--box-shadow);
}
