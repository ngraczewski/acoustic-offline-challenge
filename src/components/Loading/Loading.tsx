import React from 'react';

import './Loading.css';

export type Props = {
  message: string;
};

export const Loading = ({ message }: Props): JSX.Element => (
  <div className="loading">
    <div className="loading__message">{message}</div>
  </div>
);
