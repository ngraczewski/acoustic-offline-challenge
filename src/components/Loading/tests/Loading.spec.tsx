import React from 'react';
import faker from 'faker';
import { shallow } from 'enzyme';

import { Loading, Props } from '../Loading';

describe('Loading', () => {
  beforeEach(() => {
    faker.seed(1);
  });

  it('should render correctly', () => {
    const props: Props = {
      message: faker.lorem.sentence(),
    };

    const wrapper = shallow(<Loading {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
