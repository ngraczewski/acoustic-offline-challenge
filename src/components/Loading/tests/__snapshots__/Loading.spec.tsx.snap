// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly 1`] = `
<div
  className="loading"
>
  <div
    className="loading__message"
  >
    Repellat quisquam recusandae alias consequuntur corporis.
  </div>
</div>
`;
